server/src/api/v1/admin/get-referrer-stats.ts

```diff
@@ -0,0 +1,27 @@
+import { Request, Response } from "express";
+import { wrap } from "module";
+import { Sequelize } from "sequelize-typescript";
+import { wrapResponse } from "../../../functions/response-wrapper";
+import { Order } from "../../../models/order.model";
+
+export async function getReferrerStats(req: Request, res: Response) {
+    let success = true;
+    let result = await Order.findAll(
+        {
+            attributes: ['referrer', [Sequelize.fn('COUNT', Sequelize.col('referrer')), 'count']],
+            group: 'referrer',
+        })
+        .catch(error => {
+            success = false;
+            return [];
+        });
+
+    if (!success) {
+        return res.status(500).send(wrapResponse(false, { error: 'Database error' }));
+    }
+    if (result === []) {
+        return res.status(404).send(wrapResponse(false, { error: 'No refferrer found' }));
```

Comment:
```suggestion
        return res.status(404).send(wrapResponse(false, { error: 'No refferrer found' }));
```
Kannst du in diesem Fall nicht ein leeres [] zurück geben? Ist ja ansich kein Fehler